File: backend/models/job.py
from utils.db import db
from datetime import datetime

class Job(db.Model):
    __tablename__ = 'jobs'
    
    id = db.Column(db.Integer, primary_key=True)
    title = db.Column(db.String(200), nullable=False)
    description = db.Column(db.Text, nullable=False)
    department_id = db.Column(db.Integer, db.ForeignKey('departments.id'), nullable=False)
    status = db.Column(db.String(20), nullable=False, default='open')  # open, closed
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    
    # Relationships
    applications = db.relationship('Application', backref='job', lazy=True, cascade='all, delete-orphan')
    
    def to_dict(self):
        """Convert job to dictionary"""
        return {
            'id': self.id,
            'title': self.title,
            'description': self.description,
            'department_id': self.department_id,
            'department_name': self.department.name if self.department else None,
            'status': self.status,
            'created_at': self.created_at.isoformat(),
            'applications_count': len(self.applications)
        }
    
    def __repr__(self):
        return f'<Job {self.title} - {self.status}>'

File: frontend/lib/models/job.dart
class Job {
  final int id;
  final String title;
  final String description;
  final int departmentId;
  final String? departmentName;
  final String status;
  final String createdAt;
  final int? applicationsCount;

  Job({
    required this.id,
    required this.title,
    required this.description,
    required this.departmentId,
    this.departmentName,
    required this.status,
    required this.createdAt,
    this.applicationsCount,
  });

  factory Job.fromJson(Map<String, dynamic> json) {
    return Job(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      departmentId: json['department_id'],
      departmentName: json['department_name'],
      status: json['status'],
      createdAt: json['created_at'],
      applicationsCount: json['applications_count'],
    );
  }

  bool get isOpen => status == 'open';
}
